packages/dotnet-sdk/src/TeamsFx/Bot/TeamsBotSsoPrompt.cs

```diff
@@ -0,0 +1,188 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Extensions.Logging;
+using Microsoft.Bot.Builder.Teams;
+using Microsoft.Bot.Schema.Teams;
+using Microsoft.Bot.Schema;
+using System.Text.RegularExpressions;
+using Microsoft.Extensions.Options;
+using Microsoft.TeamsFx.Configuration;
+
+namespace Microsoft.TeamsFx;
+
+/// <summary>
+/// Creates a new prompt that leverage Teams Single Sign On (SSO) support for bot to automatically sign in user and
+/// help receive oauth token, asks the user to consent if needed.
+/// </summary>
+/// <remarks>
+/// The prompt will attempt to retrieve the users current token of the desired scopes and store it in
+/// the token store. 
+/// User will be automatically signed in leveraging Teams support of Bot Single Sign On(SSO):
+/// https://docs.microsoft.com/en-us/microsoftteams/platform/bots/how-to/authentication/auth-aad-sso-bots
+/// </remarks>
+public class TeamsBotSsoPrompt : Dialog
+{
+    private TeamsBotSsoPromptSettings _settings;
+    private readonly AuthenticationOptions _authenticationOptions;
+    private const string PersistedExpires = "expires";
+    #region Util
+    private readonly ILogger<TeamsBotSsoPrompt> _logger;
+    #endregion
+
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TeamsBotSsoPrompt"/> class.
+    /// </summary>
+    /// <param name="authenticationOptions">Authentication options filled by DI.</param>
+    /// <param name="dialogId">The ID to assign to this prompt.</param>
+    /// <param name="logger">Logger of TeamsBotSsoPrompt Class.</param>
+    /// <param name="settings">Additional OAuth settings to use with this instance of the prompt.
+    /// custom validation for this prompt.</param>
+    /// <remarks>The value of <paramref name="dialogId"/> must be unique within the
+    /// <see cref="DialogSet"/> or <see cref="ComponentDialog"/> to which the prompt is added.</remarks>
+    public TeamsBotSsoPrompt(string dialogId, TeamsBotSsoPromptSettings settings,
+        ILogger<TeamsBotSsoPrompt> logger,
+        IOptions<AuthenticationOptions> authenticationOptions) : base(dialogId)
```

Comment:
The construction of TeamsBotSsoPrompt might be a bit complex because developer need to pass in `dialogId` and `settings` so that it's not handled by ASP.NET DI. While `logger` and `authenticationOptions` are created by DI container. So, developer need to use service locator pattern to get the instance.